### PR TITLE
External domain name propagation to rancher

### PR DESCRIPTION
**Pull request by alena1108**

### CODE REVIEW

**Comment by ibuildthecloud** [comment]
Can you squash, rebase, and remove WIP if done?

**Comment by alena1108** [comment]
@ibuildthecloud done

File: resources/content/schema/base/service.json [hunk 1]
```diff
@@ -49,14 +49,11 @@
...
             }
         },
-        "previousLaunchConfig": {
-            "type": "launchConfig",
+        "domainName":{
+            "type": "string",
+            "required": false,
             "nullable": true
         },
-        "previousSecondaryLaunchConfigs": {
```
Comment (cjellick) [inline resources/content/schema/base/service.json]: Were these just superfluous? Seems unrelated to the rest of the PR

File: resources/content/schema/user/user-auth.json [hunk 2]
```diff
@@ -130,19 +130,18 @@
         "externalEvent.reportedAccountId": "r",
 
         "externalStoragePoolEvent" : "r",
-        "externalStoragePoolEvent.externalId" : "r",
```
Comment (cjellick) [inline resources/content/schema/user/user-auth.json]: You removed these bc they are not needed bc they are specified in the base type?

File: code/iaas/logic/src/main/java/io/cattle/platform/process/externalevent/ExternalEventCreate.java [hunk 3]
```diff
@@ -178,6 +196,39 @@ public void doWithLockNoResult() {
         });
     }
 
+    protected void handleServiceEvent(final ExternalEvent event, ProcessState state, ProcessInstance process) {
+        lockManager.lock(new ExternalEventLock(EXERNAL_DNS_LOCK_NAME, event.getAccountId(), event.getExternalId()),
+                new LockCallbackNoReturn() {
+                    @Override
+                    public void doWithLockNoResult() {
+                        String domainName = DataAccessor.fieldString(event, ExternalEventConstants.FIELD_DOMAIN_NAME);
+                        String serviceName = DataAccessor
```
Comment (cjellick) [inline code/iaas/logic/src/main/java/io/cattle/platform/process/externalevent/ExternalEventCreate.java]: Hey, I don't think I knew about `DataAccesor.field*()` methods. Nice. Learned something.

File: code/iaas/logic/src/main/java/io/cattle/platform/process/externalevent/ExternalEventCreate.java [hunk 4]
```diff
@@ -178,6 +196,39 @@ public void doWithLockNoResult() {
...
+                                .fieldString(event, ExternalEventConstants.FIELD_SERVICE_NAME);
+                        String stackName = DataAccessor.fieldString(event, ExternalEventConstants.FIELD_STACK_NAME);
+                        if (domainName == null || serviceName == null || stackName == null) {
+                            return;
```
Comment (cjellick) [inline code/iaas/logic/src/main/java/io/cattle/platform/process/externalevent/ExternalEventCreate.java]: It seems to me that it might worth logging something at info level (_maybe_ warn). Saying that you dind't have enough information and you're exiting.

File: resources/content/schema/base/service.json (hunk 1 above)
Comment (alena1108) [inline resources/content/schema/base/service.json]: unrelated. Removed it because its not used anywhere (was left around from some other PR, forgot to clean them up)

File: resources/content/schema/user/user-auth.json (hunk 2 above)
Comment (alena1108) [inline resources/content/schema/user/user-auth.json]: yes

File: code/iaas/logic/src/main/java/io/cattle/platform/process/externalevent/ExternalEventCreate.java (hunk 3 above)
Comment (alena1108) [inline code/iaas/logic/src/main/java/io/cattle/platform/process/externalevent/ExternalEventCreate.java]: @cjellick :)

File: code/iaas/logic/src/main/java/io/cattle/platform/process/externalevent/ExternalEventCreate.java (hunk 4 above)
Comment (alena1108) [inline code/iaas/logic/src/main/java/io/cattle/platform/process/externalevent/ExternalEventCreate.java]: sure, will add

**Comment by cjellick** [comment]
LGTM

File: code/iaas/logic/src/main/java/io/cattle/platform/process/externalevent/ExternalEventCreate.java [hunk 5]
```diff
@@ -79,6 +95,8 @@ public HandlerResult handle(ProcessState state, ProcessInstance process) {
             handleVolumeEvent(event, state, process);
         } else if (ExternalEventConstants.KIND_STORAGE_POOL_EVENT.equals(event.getKind())) {
             handleStoragePoolEvent(event, state, process);
+        } else if (ExternalEventConstants.KIND_EXTERNAL_DNS_EVENT.equals(event.getKind())) {
+            handleServiceEvent(event, state, process);
```
Comment (cjellick) [inline code/iaas/logic/src/main/java/io/cattle/platform/process/externalevent/ExternalEventCreate.java]: I know I already LGTMd, but I just realized that my k8s PR has lots of references to `exrternalServiceEvent`. Do you think you could rename this method handleDnsEvent so that we don't have to worry about confusion once both are merged?

Comment (alena1108) [inline code/iaas/logic/src/main/java/io/cattle/platform/process/externalevent/ExternalEventCreate.java]: sure
